app/controllers/glossary_terms_controller.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+# create and edit Glossary terms
+class GlossaryTermsController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :show_past,
+    :show
+  ]
+  before_action :store_location, only: [:index, :show, :destroy]
+  before_action :pass_query_params, only: [:destroy, :show_past]
```

Comment:
I'm still confused about which actions require `:store_location` or `:pass_query_params`

Reply:
Me too!!!

Reply:
**Please let me know if my revised calls to `pass_query_params` and `store_location` make sense.**
(Also should they be whitelisted instead of blacklisted?)

I revised these in 94939555ad4d9e930f63a4566cc8f4062049b477 (after re-reading @mo-nathan's comments above, an email from @pellaea, and some of the source code). 
- I eliminated `pass_query_params` because GlossaryTerms are not query-able. (At least not yet. There's a [ticket](https://www.pivotaltracker.com/story/show/167809123) to make them searchable.) 
- I applied `store_location` to all actions _except_
  - `index` because, as @mo-nathan points out above, it should never have an error.
  - `create`, `update`, and `destroy`, because (a) these actions lack views, and (b) if something goes wrong the user should be returned to the place that called those actions (`new` or `edit` for `create` and `update` respectively) rather than the action itself.

Reply:
That sounds mostly correct.

I might differ on index, but it is for practical purposes a minor point, because as you say, at least at present there are no actions you can take on that page that reasonably speaking should result in errors which would redirect you back to the index.  But there is always the theoretical possibility of adding one in the future, and we might not remember to add the store_location directive to index.